sound/soc/sof/topology.c

```diff
@@ -3197,9 +3197,17 @@ static int sof_link_load(struct snd_soc_component *scomp, int index,
 	if (!link->no_pcm) {
 		link->nonatomic = true;
 
-		/* set trigger order */
-		link->trigger[0] = SND_SOC_DPCM_TRIGGER_POST;
-		link->trigger[1] = SND_SOC_DPCM_TRIGGER_POST;
+		/*
+		 * set default trigger order for all links, such as I2S
+		 * HDA and sdw. Frontend is started before the backend
+		 * when start playback and Frontend is stopped after the
+		 * backend when stop playback. The tigger order is reverse
+		 * for Capture compared to playback.
```

Comment:
suggested edit:

set default trigger order for all links. Exceptions to the rule will be handled in sof_pcm_dai_link_fixup()
For playback, the sequence is the following:
start FE, start BE, stop BE, stop FE
For Capture the sequence is inverted
start BE, start FE, stop FE, stop BE

@ranj063 can you please review this since you modified the sequence for HDaudio. Thanks!